src/databricks/labs/ucx/workspace_access/verification.py

```diff
@@ -79,3 +79,43 @@ def verify_roles_and_entitlements(self, migration_state: MigrationState, target:
 
             assert base_group_info.roles == target_group_info.roles
             assert base_group_info.entitlements == target_group_info.entitlements
+
+
+class VerifyHasMetastore:
+    def __init__(self, ws: WorkspaceClient):
+        self.metastore_id: str | None = None
+        self.default_catalog_name: str | None = None
+        self.workspace_id: int | None = None
+        self._ws = ws
+
+    def verify_metastore(self) -> bool:
+        """
+        Verifies if a metastore exists for a metastore
+        :param cfg:
+        :return:
+        """
+        metastore_exists = self.check_metastore_existence()
+        if not metastore_exists:
+            raise MetastoreNotFoundError()
+        return True
+
+    def check_metastore_existence(self) -> bool:
```

Comment:
It's not used


Remove this method, it increases the surface area, that we don't need now

Reply:
It is used in the verify_metastore method and also unit tested (if that is what you meant by surface area)

Reply:
it's not used anywhere else

Reply:
Removing it with the fix for failing integration tests which was because of not handling the non-UC env permissions error.